### DISCUSSION TITLE
Multiple reactflow and viewport instances

### DISCUSSION THREAD

**Discussion by abhi-go** [xyflow/xyflow#3804] [Ideas]
Hello, 

I am a bit of a noob - and hence this question might be a silly, easy one. Can I create multiple instances of reactflow in the app? I want to create instances within the main window and dialogs (there could potentially be multiple levels of these). Obviously, the control rests with the dialog at the top of the stack.

Does that make sense/work?

**Comment by moklick** [comment]
Sure that's possible! The only thing you need to do is to give every flow a unique id if multiple flows are visible at once. `<ReactFlow id="flow-a" ../>` etc

**Reply by abhi-go** [reply]
Thanks. Will check it out!
